cpp/oneapi/dal/algo/subgraph_isomorphism/backend/cpu/graph.hpp

```diff
@@ -127,33 +76,70 @@ class graph {
 
     void init_bit_representation(const dal::preview::detail::topology<std::int32_t>& t);
     void init_list_representation(const dal::preview::detail::topology<std::int32_t>& t);
-
-    void init_from_list(const graph_input_list_data<Cpu>* input_list_data);
-    void init_from_bit(const graph_input_bit_data<Cpu>* input_bit_data);
+    void allocate_arrays();
 };
 
+template <typename Cpu>
+void graph<Cpu>::allocate_arrays() {
+    p_degree = allocator.allocate<std::int64_t>(vertex_count);
+    p_vertex_attribute = allocator.allocate<std::int64_t>(vertex_count);
+    p_edges_attribute = allocator.allocate<std::int64_t*>(vertex_count);
+    if (!p_degree || !p_vertex_attribute || !p_edges_attribute) {
+        throw oneapi::dal::host_bad_alloc();
+    }
+
+    for (int64_t i = 0; i < vertex_count; i++) {
+        p_edges_attribute[i] = nullptr;
+        p_degree[i] = 0;
+        p_vertex_attribute[i] = 1;
+    }
+
+    if (bit_representation) {
+        std::int64_t bit_array_size = bit_vector<Cpu>::bit_vector_size(vertex_count);
+
+        p_edges_bit = allocator.template allocate<std::uint8_t*>(vertex_count);
+        if (!p_edges_bit) {
+            throw oneapi::dal::host_bad_alloc();
+        }
```

Comment:
It's much better to extract this logic (check & throw) into inner_alloc, because it's mandatory to perform the check and even in this particular function you do it thrice.